webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1568,3 +1572,59 @@ var capvDefaultServiceAccount = regexp.MustCompile("^system:serviceaccount:svc-t
 func isCAPVServiceAccount(username string) bool {
 	return capvDefaultServiceAccount.Match([]byte(username))
 }
+
+func (v validator) validateChecks(
+	ctx *pkgctx.WebhookRequestContext,
+	newVM, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+
+	if errs := v.validateChecksPowerOn(ctx, newVM, oldVM); len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+	}
+
+	return allErrs
+}
+
+func (v validator) validateChecksPowerOn(
+	ctx *pkgctx.WebhookRequestContext,
+	newVM, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+
+	const prefix = vmopv1.CheckAnnotationPowerOn + "/"
+	var allErrs field.ErrorList
+
+	if ctx.IsPrivilegedAccount {
+		return allErrs
+	}
+
+	for k := range oldVM.Annotations {
+		if strings.HasPrefix(k, prefix) {
+			f := field.NewPath("metadata", "annotations", k)
+			_, onOldVM := oldVM.Annotations[k]
```

Comment:
Is this line needed since you're already looping through oldVM.Annotations (i.e. `onOldVM` is always true)?

Reply:
Good catch, fixed!